### ISSUE TITLE
Update Assemblies to .NET Framework 4.7.2

### ISSUE THREAD

**Issue by ahoefling** [dnnsoftware/Dnn.Platform#2615]
Part of: #2599 

## Description of problem
As DNN approaches a migration to .NET Core all the assemblies will need to implement .NET Standard 2.0 which is not supported by .NET Framework 4.5. The .NET Standard compatibility [document](https://github.com/dotnet/standard/blob/master/docs/versions.md) states that .NET Framework 4.7.2 is recommended for the best .NET Standard 2.0 compatibility.

During DNN Summit 2019 Jon Galloway of Microsoft recommended we update the platform to 4.7.2 as there aren't many significant API changes that would cause breaking changes. 

## Description of solution
Update every project in `DNN_Platform.sln` to build under .NET Framework 4.7.2

| Project | Current Version |
|---------|-------------------|
| Dnn.Modules.Console | 4.5 |
| Dnn.Modules.ModuleCreator | 4.5 |
| Dnn.AzureConnector | 4.5 |
| Dnn.GoogleAnalyticsConnector | 4.5 |
| DnnExportImport | 4.5 |
| DnnExportImportLibrary | 4.5 |
| DotNetNuke.Modules.CoreMessaging | 4.5 |
| DotNetNuke.Modules.DDRMenu | 4.5 |
| DotNetNuke.Modules.DigitalAssets | 4.5 |
| DotNetNuke.Modules.Groups | 4.5 |
| DotNetNuke.Modules.Html | 4.5 |
| DotNetNuke.Modules.HtmlEditorManager | 4.5 |
| DotNetNuke.Modules.Journal | 4.5 |
| DotNetNuke.Modules.MemberDirectory | 4.5 |
| DotNetNuke.Modules.RazorHost | 4.5 |
| DotNetNuke.Web.Deprecated | 4.5 |
| DotNetNuke.Website.Deprecated | 4.5 |
| Dnn.AuthServices.Jwt | 4.5 |
| DotNetNuke.Authentication.Facebook | 4.5 |
| DotNetNuke.Authentication.Google | 4.5 |
| DotNetNuke.Authentication.LiveConnect | 4.5 |
| DotNetNuke.Authentication.Twitter | 4.5 |
| DotNetNuke.Providers.AspNetCCP | 4.5 |
| DotNetNuke.Providers.FolderProviders | 4.5 |
| DotNetNuke.Skins.Xcillion | 4.5 |
| CountryListBox | 4.5 |
| DotNetNuke.Instrumentation | 4.5 |
| DotNetNuke.log4net | 4.5 |
| DotNetNuke.Services.Syndication | 4.5 |
| DotNetNuke.Web.Client | 4.5 |
| DotNetNuke.WebUtility | 4.5 |
| DNN.Integration.Test.Framework | 4.5 |
| DotNetNuke.Tests.Integration | 4.5 |
| DotNetNuke.Tests.Urls | 4.5 |
| DotNetNuke.Tests.AspNetCCP | 4.5 |
| DotNetNuke.Tests.Authentication | 4.5 |
| DotNetNuke.Tests.Content | 4.5 |
| DotNetNuke.Tests.Core | 4.5 |
| DotNetNuke.Tests.Data | 4.5 |
| DotNetNuke.Tests.UI | 4.5 |
| DotNetNuke.Tests.Web | 4.5 |
| DotNetNuke.Tests.Web.Mvc | 4.5 |
| DotNetNuke.Tests.Utilities | 4.5 |
| DotNetNuke.HttpModules | 4.5 |
| DotNetNuke.Library | 4.5 |
| DotNetNuke.Web | 4.5 |
| DotNetNuke.Web.Mvc | 4.5 |
| DotNetNuke.Web.Razor | 4.5 |
| DotNetNuke.Website | 4.5 |

### Servers & Admins
Once this change is implemented all servers running DNN that will want to upgrade will require downloading and installing [.NET Framework 4.7.2 Runtime](https://dotnet.microsoft.com/download/thank-you/net472)

[The .NET Download Page](https://dotnet.microsoft.com/download/visual-studio-sdks)

### Developers 
Visual Studio 2017 includes the .NET Framework 4.7.1 and developers will need to download the [4.7.2 SDK](https://dotnet.microsoft.com/download/visual-studio-sdks) to be continue working.

Visual Studio 2019 is quickly approaching a release date and that should include .NET Framework 4.7.2

## Additional context
Add any other context about the feature that may be helpful with implementation.

## Affected version
Planning on implementing this change for DNN 9.4.0

* [x] 9.4.0

**Comment by ohine** [comment]
v4.7.2 upgrade is pending in PR #2144

**Comment by ahoefling** [comment]
Perfect! Thanks for linking it.

**Comment by mitchelsellers** [comment]
Additional note, in confirming with hosting provides and cloud solution providers the .NET 4.7.2 updates seem to be getting installed in most environments automatically.

I have checked with multiple major hosting companies to inquire on their shared & managed environments and they state that all have .NET 4.7.2 installed.  Additionally, Microsoft Azure App Service by default is running .NET 4.7.2.

Per Microsoft the only changes between version that impact us are either Edge or Minor, which would be fitting in support of inclusion in 9.4.0. [Microsoft Change History](https://docs.microsoft.com/en-us/dotnet/framework/migration-guide/runtime/4.5-4.7.2)

**Comment by ahoefling** [comment]
With DNN 9.3.0 released is there a plan on when we want to get the 4.7.2 PR merged into the platform? I would like to start analyzing a port of modules and libraries to .NET Standard 2.0 and this step needs to be completed first.

cc: @mitchelsellers @ohine

**Comment by mitchelsellers** [comment]
We can merge at anytime. Just need to fix the merge issues.

**Comment by ahoefling** [comment]
Per discussion in #2144 I'm going to open a new PR to close this issue out
